cdap-ui/app/cdap/components/Experiments/store/ActionCreator.js

```diff
@@ -159,79 +160,177 @@ function setWorkspace(workspaceId) {
+      });
+    });
 }
 
-function createExperimentAndModel() {
-  let {experiments_create, model_create} = createExperimentStore.getState();
-  let {selectedNamespace: namespace} = NamespaceStore.getState();
-  let {workspaceId, directives} = model_create;
+function createExperiment() {
+  let {model_create, experiments_create} = createExperimentStore.getState();
+  let {directives} = model_create;
   let requestBody = directiveRequestBodyCreator(directives);
   let experiment = {
-    name: experiments_create.name,
-    description: experiments_create.description,
-    outcome: experiments_create.outcome,
-    srcpath: experiments_create.srcpath,
-    workspaceId: model_create.workspaceId
-  };
-  setExperimentLoading();
-  let fields;
-
-  let model = {
-    name: model_create.name,
-    description: model_create.description,
-    algorithm: model_create.algorithm.name,
-    hyperparameters: {},
-    split: 'random',
-    directives: model_create.directives,
-    features: model_create.columns.filter(column => column !== experiments_create.outcome)
+    ...experiments_create
   };
-
   MyDataPrepApi
-    .getSchema({ namespace, workspaceId}, requestBody)
+    .getSchema({
+      namespace: getCurrentNamespace(),
+      workspaceId: experiments_create.workspaceId
+    }, requestBody)
     .mergeMap((schema) => {
-      fields = schema;
       // The outcome will always be simple type. So ["null", "anything"] should give correct outcomeType at the end.
       let outcomeType = schema
         .find(field => field.name === experiment.outcome)
         .type
         .filter(t => t !== 'null')
         .pop();
       experiment.outcomeType = outcomeType;
-      return createExperiment(experiment);
+      updateSchema(schema);
+      return myExperimentsApi.createExperiment({
+        namespace: getCurrentNamespace(),
+        experimentId: experiment.name
+      }, experiment);
     })
-    .mergeMap(() => {
-      let tempSchema = {
-        name: 'avroSchema',
-        type: 'record',
-        fields
-      };
-      let splitInfo = {
-        schema: tempSchema,
-        directives,
-        type: 'random',
-        parameters: { percent: "80"},
-        description: `Default Random split created for model: ${model_create.name}`
-      };
-      return myExperimentsApi.createSplit({namespace, experimentId: experiments_create.name}, splitInfo);
+    .subscribe(setExperimentCreated.bind(null, experiments_create.name));
+}
+
+function pollForSplitStatus(experimentId, split) {
+  let count = 1;
+  const getStatusOfSplit = (callback, errorCallback, count) => {
+    const params = {
+      namespace: getCurrentNamespace(),
+      experimentId,
+      splitId: split.id
+    };
+    myExperimentsApi
+      .getSplitStatus(params)
+      .subscribe(status => {
+        if (status !== 'Complete') {
+          if (count < 120) {
+            count += count;
+            setTimeout(() => {
+              getStatusOfSplit(callback, errorCallback, count);
```

Comment:
Just curious, if we don't pass the `count` to `getStatusOfSplit()`, couldn't we still use and update `count`, because it's defined in the lexical scope of the function? (at line 223)